notebooks/04-Conditionals.ipynb

```diff
@@ -0,0 +1,1275 @@
+     "slide_type": "-"
+    }
+   },
+   "source": [
+    "   ```python\n",
+    "   if tired == 'Yes':\n",
+    "       go_to_bed()\n",
+    "   else:\n",
+    "       read_a_book()\n",
+    "   ```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "source": [
+    "* Similar to `if`'s then-block, there's also an indented block to be run if the `else` path is taken"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "<div class='question'>\n",
+    "    <strong>Question:</strong> When will the <code>else</code> path be taken?\n",
+    "</div>\n",
+    "<br>\n",
+    "\n",
+    "   ```python\n",
+    "   if tired == 'Yes':\n",
+    "       go_to_bed()\n",
+    "   else:\n",
+    "       read_a_book()\n",
+    "   ```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "* The `else` path is taken in all cases when `tired` is not equal to `'Yes'`"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### Your Turn\n",
+    "\n",
+    "Fill in the blanks to print the relevant statement for `x`. Re-run multiple times.\n",
+    "\n",
+    "```python\n",
+    "import random\n",
+    "\n",
+    "x = random.randint(-10, 10)\n",
+    "\n",
+    "__ x _ __:\n",
+    "    print('x is negative')\n",
+    "__:\n",
+    "    print('x is positive')\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### `elif` Statement"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "* There's another option available when working with conditionals in Python"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "* The `elif` (else-if) statement can be used to add another `if` statement in the execution path"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "-"
+    }
+   },
+   "source": [
+    "   ```python\n",
+    "   if tired == 'Yes':\n",
+    "       go_to_bed()\n",
+    "   elif tired == 'A little':\n",
+    "       rest_eyes()\n",
+    "   else:\n",
+    "       read_a_book()\n",
+    "   ```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "* You can include as many `elif` statements as you want\n",
+    "    * if you're familiar with a `switch` statement in other languages, this is Python's closest relative"
```

Comment:
Do we want comments (not on the slide) about the pattern matching available in 3.10? We are using it for the course, although many corporate environments might still be somewhere between 3.7 to 3.9.

Reply:
I didn't realize we were using it elsewhere. I'll plan to mention it in that case, and if I have time I'll come back and add a slide for it.